Kinvey.Core/Realtime/Stream.cs

```diff
@@ -213,9 +214,9 @@ public Stream(string streamName, AbstractClient client = null)
 			}
 
 			// Make KCS request for subscribe access to a user with the given subscribeID
-			success = await RequestSubscribeAccess(subscribeID);
+			var result = await RequestSubscribeAccess(subscribeID);
 
-			if (success)
+			if (result != null)
```

Comment:
@tejasranade I believe this could be `null` if the deserialization of the response content is not successful.  In this case, the type returned is `default(JObject)`, which is null.  But I can change this to check it against `default(JObject)`, rather than `null`.

As for server exceptions, they are caught and thrown up the stack just as in 4xx/5xx responses from other endpoint requests.